### ISSUE TITLE
Using DateTime in UrlEncoded Refit Interfaces

### ISSUE THREAD

**Issue by martingemme** [paulcbetts/refit#220]
I want to use "2016/02/15" in URL from DateTime using the [Zimbra Api](https://wiki.zimbra.com/wiki/Zimbra_REST_API_Reference:Get_Calendar).

Based on https://github.com/paulcbetts/refit/issues/64, it suggests to use the following pattern:

``` c#
public class ZimbraUrlParameterFormatter : DefaultUrlParameterFormatter
{
    public override string Format(object parameterValue, ParameterInfo parameterInfo)
    {
        if (parameterValue == null)
            return null;

        // Look for a Date
        var date = parameterValue as DateTime?;
        if (date.HasValue)
        {
            return date.Value.ToString("yyyy/MM/dd", CultureInfo.InvariantCulture);
        }

        return base.Format(parameterValue, parameterInfo);
    }
}
```

The Refit interface looks a bit like this

``` c#
public interface ICalendarProvider
{
    [Post("/home/{username}/calendar?start={start}&end={end}&fmt=json&auth=ba")]
    Task<RootObject> GetScheduleFor(string username, DateTime start, DateTime end);
}
```

But however looking in fiddler, the slashes are encoded i.e. 2016%2f02%2f15.

How can I directly format the datetime into "yyyy/MM/dd" bypassing the encoding?

**Comment by bennor** [comment]
The url encoding isn't configurable at the moment. So the only options are a) submit a PR that makes it configurable, or b) look for a different format.

_Seems really odd that they'll only accept an incorrectly encoded querystring parameter._

**Comment by martingemme** [comment]
I also have another example. The **Microsoft Graph Api**, namely:

`
public interface IOffice365Api
{
   // This query gets the calendar events 'from' to 'to' in ToString("s") format        
   // https://graph.microsoft.com/v1.0/me/calendar/events
   // ?$top=100&$filter=Start/DateTime ge '2017-05-01T12:00:00Z' and End/DateTime lt '2017-05-05T12:00:00Z'
   [Get("/me/calendar/events?$top=100&$filter=Start/DateTime ge '{startTime}Z' and End/DateTime lt '{endTime}Z'")]
   Task<OutlookRoot<OutlookEvent>> GetEvents(string startTime, string endTime);        
}
`

It really should be (for example)
https://graph.microsoft.com/v1.0/me/calendar/events?$top=100&$filter=Start/DateTime ge '2017-05-30T00:00:00Z' and End/DateTime lt '2017-06-05T00:00:00ZZ'
But it ends up being:
https://graph.microsoft.com/v1.0/me/calendar/events?$top=100&$filter=Start%2FDateTime ge '2017-05-30T00:00:00Z' and End%2FDateTime lt '2017-05-05T00:00:00Z'

**Comment by onovotny** [comment]
Fixed in #415
